sys/include/net/hdlc/hdr.h

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright (C) 2015 José Ignacio Alamos <jialaos@uc.cl>
+ * Copyright (C) 2018 Max van Kessel
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser General
+ * Public License v2.1. See the file LICENSE in the top level directory for
+ * more details.
+ */
+
+/**
+ * @defgroup    net_hdlc_hdr    HDLC header
+ * @ingroup     net_hdlc
+ * @brief       HDLC header architecture
+ *
+ * @{
+ *
+ * @file
+ * @brief       Definitions for High Level Data Link Control header
+ *
+ * @author      José Ignacio Alamos <jialamos@uc.cl>
+ * @author      Max van Kessel
+ */
+#ifndef NET_HDLC_HDR_H
+#define NET_HDLC_HDR_H
+
+#include <stdint.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief HDLC types
+ */
+typedef enum {
+    HDLC_TYPE_RECEIVE_READY     = 0,        /**< ready receive type */
+    HDLC_TYPE_RECEIVE_NOT_READY = (1 << 1), /**< receive not ready */
+    HDLC_TYPE_REJECT            = (1 << 0), /**< reject */
+    HDLC_TYPE_SELECTIVE_REJECT  = (3 << 0), /**< selective reject */
+} hdlc_type_t;
+
+/**
+ * @brief HDLC Frame types
+ */
+typedef enum {
+    HDLC_FRAME_TYPE_INFORMATION = 0, /**< represents i(nformation) frame */
+    HDLC_FRAME_TYPE_SUPERVISORY = 1, /**< represents s(upervisory) frame*/
+    HDLC_FRAME_TYPE_UNNUMBERED = 3,  /**< represents u(numbered) frame */
+} hdlc_frame_type_t;
+
+/**
+ * @brief HDLC Supervisory Frames
+ */
+typedef struct __attribute__((packed)) {
```

Comment:
Please don't use bitfields for wire formats. The actual order is implementation defined and thus cannot be relied upon.